substrate/frame/revive/src/wasm/runtime.rs

```diff
@@ -1324,6 +1327,41 @@ pub mod env {
 		}
 	}
 
+	/// Stores the U256 value at given call input `offset` into the supplied buffer.
+	/// See [`pallet_revive_uapi::HostFn::call_data_load`].
+	#[api_version(0)]
+	fn call_data_load(
+		&mut self,
+		memory: &mut M,
+		out_ptr: u32,
+		offset: u32,
+	) -> Result<(), TrapReason> {
+		self.charge_gas(RuntimeCosts::CallDataLoad)?;
+
+		let Some(input) = self.input_data.as_ref() else {
+			return Err(Error::<E::T>::InputForwarded.into());
+		};
+
+		let mut data = [0; 32];
+		let len = input.len();
+		let start = offset as usize;
+		let data = if start > len {
+			data // Any index is valid to request; OOB offsets return zero.
+		} else {
+			let mut end = start.saturating_add(32);
+			if end > len {
+				end = len;
+			}
```

Comment:
can't we write it like this 
let end = (start + 32).min(len);

Reply:
Yes, thanks